### DISCUSSION TITLE
datagrid not showing

### DISCUSSION THREAD

**Discussion by kuku20009** [AvaloniaUI/Avalonia#13796] [Q&A]
## Describe the bug
datagrid not showing
![image](https://github.com/AvaloniaUI/Avalonia/assets/39628682/20f8b721-bb23-454a-8e26-e5d79e0fdd3a)

with all lib version: 11.0.5
![image](https://github.com/AvaloniaUI/Avalonia/assets/39628682/fa46f25f-6235-4535-b300-0ad9647834b5)

## To Reproduce
write demo from :https://docs.avaloniaui.net/docs/reference/controls/datagrid/

1. add app.axaml
```xaml
<Application.Styles>
        <FluentTheme />
		<StyleInclude Source="avares://Avalonia.Controls.DataGrid/Themes/Fluent.xaml"/>
    </Application.Styles>
```
2. ui
```xaml
<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:vm="using:AvaloniaApplication1.ViewModels"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        xmlns:views="clr-namespace:AvaloniaApplication1.Views"
        mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
        x:Class="AvaloniaApplication1.Views.MainWindow"
        Icon="/Assets/avalonia-logo.ico"
        Title="AvaloniaApplication1">
	<DataGrid Margin="20" ItemsSource="{Binding People}"
	  AutoGenerateColumns="True" IsReadOnly="True"
	  GridLinesVisibility="All"
	  BorderThickness="1" BorderBrush="Gray">
	</DataGrid>
</Window>
```

3.code
```csharp
using Avalonia.Controls;
using System.Collections.ObjectModel;
using System;
using System.Collections.Generic;

namespace AvaloniaApplication1.Views;

public partial class MainWindow : Window
{
    public ObservableCollection<Person> People { get; }
    public MainWindow()
    {
        var people = new List<Person>
            {
                new Person("Neil", "Armstrong"),
                new Person("Buzz", "Lightyear"),
                new Person("James", "Kirk")
            };
        People = new ObservableCollection<Person>(people);
        InitializeComponent();
    }
}

public class Person
{
    public string FirstName { get; set; }
    public string LastName { get; set; }

    public Person(string firstName, string lastName)
    {
        FirstName = firstName;
        LastName = lastName;
    }
}
```

## Expected behavior
show table

## Environment 
vs2022

- OS: windows
- Avalonia-Version: 11.0.5

**Comment by timunie** [comment]
package is installed?

**Comment by timunie** [comment]
Also where do you set the DataContext of the Window? It should be set to self somewhere

**Comment by thevortexcloud** [comment]
Is that a typo or are you actually missing the theme for the datagrid?

**Comment by MrJul** [comment]
I've edited the original post with code formatting so it's clearer.

The datagrid is displaying, see the gray border showing.
You didn't set `Window.DataContext`: the `People` property isn't found and no columns nor rows are generated.

(In the sample, the `People` collection is in its own `MainViewModel`, not `MainWindow`).

I'm converting this issue to a discussion.

**Comment by kuku20009** [comment]
3Q everyone, new to avalonia. it is just a mistake.
after reading:https://docs.avaloniaui.net/docs/basics/data/data-binding/data-context

i know what happens. then show it :
![image](https://github.com/AvaloniaUI/Avalonia/assets/39628682/9f2ac758-767a-458e-adfa-4f6d642cc03f)
